### PR TITLE
{chem}[foss/2023a,intel/2023a] Wannier90 v3.1.0

### PR DESCRIPTION
**Pull request by schiotz**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by schiotz** [comment]
Test report by @schiotz
**SUCCESS**
Build succeeded for 8 out of 8 (2 easyconfigs in total)
sylg.fysik.dtu.dk - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2650 v4 @ 2.20GHz, Python 3.6.8
See https://gist.github.com/schiotz/275a549613b9044b6dc88c582818a42d for a full test report.

File: easybuild/easyconfigs/w/Wannier90/Wannier90-3.1.0-intel-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+sources = [{'download_filename': 'v%(version)s.tar.gz', 'filename': SOURCELOWER_TAR_GZ}]
+patches = ['Wannier90_3x_ignore_makeinc.patch']
+checksums = [
+    '40651a9832eb93dec20a8360dd535262c261c34e13c41b6755fa6915c936b254',  # wannier90-3.1.0.tar.gz
+    '561c0d296e0e30b8bb303702cd6e41ded54c153d9b9e6cd9cab73858e5e2945e',  # Wannier90_3x_ignore_makeinc.patch
+]
+
+# The -fallow-argument-mismatch allows MPI communication calls to be
+# called with arrays of different types at different places in the
+# code.  This otherwise cause an error in GCC 10.X
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/w/Wannier90/Wannier90-3.1.0-intel-2023a.eb:23]: I think this commend only applies to the foss toolchain which uses GCC.
```suggestion
```

**Review by SebastianAchilles — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/w/Wannier90/Wannier90-3.1.0-intel-2023a.eb (hunk 1 above)
Comment (schiotz) [reply easybuild/easyconfigs/w/Wannier90/Wannier90-3.1.0-intel-2023a.eb:23]: Thank you @SebastianAchilles for catching this.  Fix is incoming...

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.9.16
See https://gist.github.com/SebastianAchilles/56f281e206a958ee684b9a2793a48c71 for a full test report.

**Comment by SebastianAchilles** [comment]
Going in, thanks @schiotz!

**Pull request merged**